src/query/src/range_select/plan_rewrite.rs

```diff
@@ -104,23 +109,22 @@ impl<'a> TreeNodeRewriter for RangeExprRewriter<'a> {
     fn mutate(&mut self, node: Expr) -> DFResult<Expr> {
         if let Expr::ScalarUDF(func) = &node {
             if func.fun.name == "range_fn" {
-                // `range_fn(func_name, argc, [argv], range, fill, byc, [byv], align)`
-                // `argsv` and `byv` are variadic arguments, argc/byc indicate the length of arguments
-                let func_name = parse_str_expr(&func.args, 0)?;
-                let argc = str::parse::<usize>(parse_str_expr(&func.args, 1)?)
+                // `range_fn(func, range, fill, byc, [byv], align)`
+                // `[byv]` are variadic arguments, byc indicate the length of arguments
+                let range_expr = self.get_range_expr(&func.args, 0)?;
+                let range_str = parse_str_expr(&func.args, 1)?;
+                let byc = str::parse::<usize>(parse_str_expr(&func.args, 3)?)
                     .map_err(|e| DataFusionError::Plan(e.to_string()))?;
-                let byc = str::parse::<usize>(parse_str_expr(&func.args, argc + 4)?)
-                    .map_err(|e| DataFusionError::Plan(e.to_string()))?;
-                let mut range_fn = RangeFn {
-                    expr: Expr::Wildcard,
-                    range: parse_duration(parse_str_expr(&func.args, argc + 2)?)
-                        .map_err(DataFusionError::Plan)?,
-                    fill: parse_str_expr(&func.args, argc + 3)?.to_string(),
-                };
-                let args = parse_expr_list(&func.args, 2, argc)?;
-                let by = parse_expr_list(&func.args, argc + 5, byc)?;
-                let align = parse_duration(parse_str_expr(&func.args, argc + byc + 5)?)
+                let by = parse_expr_list(&func.args, 4, byc)?;
+                let align = parse_duration(parse_str_expr(&func.args, byc + 4)?)
                     .map_err(DataFusionError::Plan)?;
+                let mut data_type = range_expr.get_type(self.input_plan.schema())?;
+                let mut need_cast = false;
+                let fill = Fill::try_from_str(parse_str_expr(&func.args, 2)?, &data_type)?;
```

Comment:
suggestion: looks like the `parse_str_expr` function doesn't contain the wrong exp in the error message when it's not a string expr. I think we can improve these error messages to help users.

Reply:
I think some more detailed information can be added, but if the user directly uses our range query without knowing the implementation logic of the range query, the errors that emerge during the rewriting phase are likely to confuse the user.

Reply:
Yes, but for some errors such as invalid duration, we can provide a detailed error message to help users debug by themselves.